src/frontend/utils/chat/chat.ts

```diff
@@ -0,0 +1,31 @@
+import { converse } from 'utils/window';
+
+export const generateAnonymousNickname = (): string => {
+  const tokenLength = 8;
+  const charUniverse =
+    'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'.split('');
+  const generatedToken = [];
+  for (let i = 0; i < tokenLength; i++) {
+    const j = Math.round(Math.random() * (charUniverse.length - 1));
+    generatedToken[i] = charUniverse[j];
+  }
+  return 'Anonymous-'.concat(generatedToken.join(''));
```

Comment:
```suggestion
const generatedToken =  [...Array(tokenLength).keys()].map(() => {
    return charUniverse[Math.round(Math.random() * (charUniverse.length - 1))];
  }).join('');

  return `Anonymous-${generatedToken}`;
```

Reply:
done in [94260d](https://github.com/openfun/marsha/pull/1276/commits/94260daa50c03f34070086bde57fb2de9a43b5f8)